R/plotPValHist.R

```diff
@@ -16,63 +17,78 @@
 #' }
 #'
 #' @import ggplot2
 #' @importFrom dplyr filter select
 #' @importFrom canvasXpress canvasXpress
 #'
 #' @export
-plotPvalHist <- function(P.Val,
-                         plotType = "canvasXpress",
+plotPvalHist <- function(DGEdata,
+                         P.Val          = "P.Value",
+                         plotType       = "canvasXpress",
                          facet          = TRUE,
                          binWidth       = 0.02,
                          transparency   = 0.6,
                          color    = "dodgerblue3") {
 
-    plotType = tolower(plotType)
+    assertthat::assert_that(!missing(DGEdata),
+                            !is.null(DGEdata),
+                            "DGEobj" %in% class(DGEdata),
+                            msg = "DGEdata must be specified and must belong to DGEobj class.")
 
-    assertthat::assert_that(!missing(P.Val),
-                            class(P.Val)[[1]] %in% c("matrix","data.frame"),
-                            all(sapply(data.frame(P.Val),is.numeric)),
-                            msg = "P.Val must be specified and must be of class matrix or dataframe and must contain only numeric values.")
+    if (any(is.null(P.Val),
+            !is.character(P.Val),
+            length(P.Val) != 1)) {
+        warning("P.Val must be a singular value of class character. Assigning default value 'P.Value'.")
+        P.Val <- "P.Value"
+    }
 
-    assertthat::assert_that(all(sapply(data.frame(P.Val),is.numeric)),
-                            msg = "P.Val must contain only numeric values.")
+    P.Val <- extractCol(getType(DGEdata, "topTable"), colName = P.Val, robust = FALSE)
 
-    assertthat::assert_that(plotType %in% c("canvasxpress", "ggplot"),
-                            msg = "Plot type must be either canvasXpress or ggplot.")
+    plotType <- tolower(plotType)
+    if (any(is.null(plotType),
+            !is.character(plotType),
+            length(plotType) != 1,
+            !plotType %in% c("canvasxpress", "ggplot"))) {
+        warning("plotType must be either canvasXpress or ggplot. Assigning default value 'CanvasXpress'.")
+        plotType <- "canvasxpress"
+    }
 
-    if (!assertthat::see_if(is.character(color),
-                            length(color) == 1)) {
+    if (any(is.null(color),
+            !is.character(color),
+            length(color)  != 1,
+            length(.validate_colors(color)) != 1)) {
         warning("color must be a singular value of class character and must specify the name of the color or the rgb value. Assigning default value 'dodgerblue3'.")
         color <- "dodgerblue3"
     }
 
-    if (!assertthat::see_if(is.logical(facet),
-                            length(facet) == 1)) {
+    if (any(is.null(facet),
+            !is.logical(facet),
+            length(facet) != 1)) {
         warning("facet must be a singular logical value. Assigning default value TRUE.")
-        facet <- TRUE
+        facet = TRUE
```

Comment:
Why using "=" sign here instead of the usual `<-`?

Reply:
changed.